service/src/test/java/uk/gov/hmcts/reform/fpl/jobs/UndeliveredEmailsFinderTest.java

```diff
@@ -0,0 +1,168 @@
+package uk.gov.hmcts.reform.fpl.jobs;
+
+import org.joda.time.DateTime;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.quartz.JobDetail;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobKey;
+import org.springframework.context.ApplicationEventPublisher;
+import uk.gov.hmcts.reform.fpl.events.UndeliveredEmailsFound;
+import uk.gov.hmcts.reform.fpl.exceptions.JobException;
+import uk.gov.hmcts.reform.fpl.model.UndeliveredEmail;
+import uk.gov.hmcts.reform.fpl.utils.extension.TestLogger;
+import uk.gov.hmcts.reform.fpl.utils.extension.TestLogs;
+import uk.gov.hmcts.reform.fpl.utils.extension.TestLogsExtension;
+import uk.gov.service.notify.Notification;
+import uk.gov.service.notify.NotificationClient;
+import uk.gov.service.notify.NotificationList;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith({MockitoExtension.class, TestLogsExtension.class})
+class UndeliveredEmailsFinderTest {
+
```

Comment:
could use the Time instead of static Datetime with fixedTime to test the edge second rather than setting longer in the past/future

Reply:
it is joda time